applications/asset_tracker/src/env_sensors/env_sensors.c

```diff
@@ -214,8 +214,8 @@ void env_sensors_set_send_interval(const uint32_t interval_s)
 	if (data_send_interval_s) {
 		/* restart work for new interval to take effect */
 		env_sensors_poll();
-	} else if (k_delayed_work_remaining_get(&env_sensors_poller) > 0) {
-		k_delayed_work_cancel(&env_sensors_poller);
+	} else {
+		k_work_cancel_delayable(&env_sensors_poller);
```

Comment:
Please document in a comment what happens if this does not complete immediately so the work continues to run.